src/views/htmlcontent/src/js/components/app.component.ts

```diff
@@ -468,14 +470,47 @@ export class AppComponent implements OnInit, AfterViewChecked {
 
     openContextMenu(event: {x: number, y: number}, batchId, resultId, index): void {
         let selection = this.slickgrids.toArray()[index].getSelectedRanges();
+        selection = this.tryCombineSelections(selection);
         this.contextMenu.show(event.x, event.y, batchId, resultId, index, selection);
     }
 
+    private tryCombineSelections(selections: ISlickRange[]): ISlickRange[] {
```

Comment:
How expensive is this? E.g. any chance this'll take more than a few milliseconds? Just want to ensure we don't hold up the UI responsiveness with pre-processing logic.

Reply:
I tested it out on some queries with pretty big result sets and it performed well, but I'll double check